### ISSUE TITLE
Not loading matches in firefox

### ISSUE THREAD

**Issue by IsXGeo** [ge-ku/Ban-Checker-for-Steam#16]
Hello! Thanks for the awesome plugin!
I am using Firefox Quantum 60.0.1 (64-bit) and plugin version 1.1.5,
After adding my API key, which works successfully, I ran the plugin yet it only showed a total of eight games. It didn't change for about 10 minutes when I gave up. I tried with chrome and it worked successfully.

https://imgur.com/a/W7doHuZ

**Comment by ge-ku** [comment]
Simply loading matched does not use or require API key, I'm not sure why it doesn't load whole match history in Firefox for you. Can you please check if there are any errors in console when you try to replicate this?

**Comment by LeoVerto** [comment]
I'm having the same problem and there's nothing in the console other than the standard
`` Continue token: XXX | SessionID: XXX | Profile: https://steamcommunity.com/id/xxx/

**Comment by ge-ku** [comment]
Please try again after updating to 1.1.7. It should give better error message.

**Comment by LeoVerto** [comment]
I'm getting an "Error: error parsing JSON" with the hint

    "<!DOCTYPE html>\r\n<html class=\" responsive\" lang=\"en\">\r\n<head>\r\n\t<meta http-equiv=\"Content-Type\" content=\"text/html; charset=UTF-8\">\r\n\t\t\t<meta name=\"viewport\" content=\"width=device-width,initial-scale=1\">\r\n\t\t<meta name=\"theme-color\" content=\"#171a21\">\r\n\t\t<title>Steam Community :: Leo Verto</title
and so on, which seems to be the entire HTML source of my profile page.

**Comment by ge-ku** [comment]
Check again using 1.1.7.1, now it should show which request caused this error, I hope it leads somewhere as currently I have no idea why it happens.

**Comment by LeoVerto** [comment]
Looks like it requested ``https://steamcommunity.com/id/leo_verto/gcpd/730?ajax=1&tab=matchhistorycompetitive&continue_token=XXX`` which when manually opened seemed to be valid JSON in the format

    {
      "success":true,
      "html":"<table class=\"generic_kv_table csgo_scoreboard_root\">\r\n<tr>\r\n\t<th class=\"col_left\">Map<\/th>\r\n\t<th>Match Results<\/th>[SHORTENED]\r\n<\/table>\r\n",
      "continue_token":"YYY"
    }

I didn't see that request in the developer console, could this be a forced redirect say because Firefox doesn't include cookies in the header of the Addon's fetch()?

**Comment by ge-ku** [comment]
This is likely it (though it worked for me as is), I just uploaded new version (only for Firefox for now), version is 1.1.7.2 where `credentials` is set to `include` instead of `same-origin`. Can you please check if it works now?

**Comment by ge-ku** [comment]
Finally found how to reproduce this, if I set ["Accept third-party cookies"](https://prod-cdn.sumo.mozilla.net/uploads/gallery/images/2018-03-26-06-46-23-115200.png) to Never it gives me this same result. I also found [this topic on Mozilla Discourse](https://discourse.mozilla.org/t/webextension-xmlhttprequest-fetch-cookies-are-not-sent-even-with-third-party-cookies-allowed/28581) where someone had this issue even with third-party cookies enabled until they launched Firefox with clean profile. Still looking for a solution.
